frontend/src/metabase/redux/embed/embed.unit.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { type Dispatch, configureStore } from "@reduxjs/toolkit";
+
+import {
+  DEFAULT_INTERACTIVE_EMBEDDING_OPTIONS,
+  embed as embedReducer,
+  setInitialUrlOptions,
+} from "./embed";
+
+describe("embed reducer", () => {
+  describe("setInitialUrlOptions", () => {
+    it("should set default options", () => {
+      const store = createMockStore();
+
+      store.dispatch(setInitialUrlOptions({ search: "" }));
+
+      expect(store.getState().embed.options).toEqual(
+        DEFAULT_INTERACTIVE_EMBEDDING_OPTIONS,
+      );
+    });
+
+    it("should set options from search", () => {
+      const store = createMockStore();
+
+      store.dispatch(
+        setInitialUrlOptions({ search: "top_nav=false&new_button=true" }),
+      );
+
+      expect(store.getState().embed.options.top_nav).toBe(false);
+      expect(store.getState().embed.options.new_button).toBe(true);
+    });
+
+    it("should ignore invalid options", () => {
+      const store = createMockStore();
+
+      store.dispatch(
+        setInitialUrlOptions({ search: "top_nav=false&invalid_option=123" }),
+      );
+
+      expect(store.getState().embed.options).not.toHaveProperty(
+        "invalid_option",
+      );
+    });
+
+    describe("entity_types", () => {
```

Comment:
https://www.notion.so/metabase/Testing-Let-embedders-hide-tables-1ae69354c901807794d2f232d701f037?pvs=4#1b369354c9018074ace3d3ba13070091

Reply:
Looks like in the above document we support both of these scenarios:

```
- comma-separated e.g. `entity_types=model,table`
- multiple `entity_types` e.g. `entity_types=table&entity_types=model`
```

In the usage docs, I would encourage people to use comma-separated form since that is less confusing (I think many people don't even know that URL parameters doesn't have to be unique)

Reply:
true, multiple params are the web standard, though. I think supporting it isn't bad, and it's a by product anyway. We won't be listing it in the doc. As the project only says we support comma-separated value.